### PR TITLE
Initial sale code review 1

### PR DESCRIPTION
**Pull request by logicalmechanism**
This PR contains a working fork of the most up-to-date implementation of the initial sale contract. This includes the pointer nft minter and the band lock-up features. With the updates comes a new batcher, batcher v4. This is a reduced version of batcher v3 but purpose-built just for the initial sale contract like batcher v2 but with all the updates from v3+. The order book is not complete.

The contracts will be integrated into the existing system by updating the data on the reference contract.

### CODE REVIEW

File: aiken-contracts/fractional/scripts/00_createScriptReferences.sh [hunk 1]
```diff
@@ -152,7 +152,7 @@ ${cli} transaction build-raw \
     --tx-out-reference-script-file ${cip68_script_path} \
     --fee 900000
 
-FEE=$(cardano-cli transaction calculate-min-fee --tx-body-file ./tmp/tx.draft --testnet-magic ${testnet_magic} --protocol-params-file ./tmp/protocol.json --tx-in-count 0 --tx-out-count 0 --witness-count 1)
+FEE=$(cardano-cli transaction calculate-min-fee --tx-body-file ./tmp/tx.draft --testnet-magic ${testnet_magic} --protocol-params-file ./tmp/protocol.json --tx-in-count 1 --tx-out-count 1 --witness-count 1)
```
Comment (AndrewWestberg) [inline aiken-contracts/fractional/scripts/00_createScriptReferences.sh:155]: We'll need to run through all these fee calculations with the next version of the node. George from MLabs has modified the fee calculations so they're actually correct.  I believe the --tx-in-count and --tx-out-count don't do anything anymore. We'll need to re-test once this gets merged and into a release: https://github.com/IntersectMBO/cardano-cli/pull/534

File: aiken-contracts/fractional/scripts/batcher_v3/src/queue_purchase.py [hunk 2]
```diff
@@ -26,35 +27,37 @@ def build_tx(sale_info, queue_info, batcher_info, constants: dict) -> Tuple[dict
     batcher_pkh = constants['batcher_pkh']
 
     # HARDCODE FEE FOR NOW, NEED WAY TO ESITMATE THESE UNITS BETTER
-    FEE = 600000
+    FEE = 505549
```
Comment (AndrewWestberg) [inline aiken-contracts/fractional/scripts/batcher_v3/src/queue_purchase.py:30]: We should be able to calculate these directly when we build the rust batcher.

**Review by AndrewWestberg — APPROVED** [review]

File: aiken-contracts/fractional/scripts/00_createScriptReferences.sh (hunk 1 above)
Comment (logicalmechanism) [reply aiken-contracts/fractional/scripts/00_createScriptReferences.sh:155]: I'll keep an eye on the PR. Seems like it should be merged anytime now. We can redo all these fee calculations after that.

File: aiken-contracts/fractional/scripts/batcher_v3/src/queue_purchase.py (hunk 2 above)
Comment (logicalmechanism) [reply aiken-contracts/fractional/scripts/batcher_v3/src/queue_purchase.py:30]: Nice! Yeah, hard coding for now as it is a limitation of my [tx simulation](https://github.com/logicalmechanism/tx_simulation) when tx chaining. I did do proper tests and the fee for the swap is more like 400,000 Lovelace and the refund is around 300,000 Lovelace so being able to calculate the real values will be worth it.

**Pull request merged**